### ISSUE TITLE
Mutliple response.WriteHeader Calls

### ISSUE THREAD

**Issue by gsquire** [linkedin/Burrow#218]
In the burrow logs, I am seeing many lines like the following:
```sh
2017/05/03 17:18:06 http: multiple response.WriteHeader calls
```
I think this is due to the fact that some HTTP handlers do not properly return in an error case. I can submit a PR to help remediate these.

**Comment by toddpalino** [comment]
I'll take a look at this as part of the rewrite

**Comment by toddpalino** [comment]
Resolved by 1.0 rewrite
